spec.html

```diff
@@ -6949,22 +6949,23 @@ <h1>ValidateAndApplyPropertyDescriptor ( _O_, _P_, _extensible_, _Desc_, _curren
             1. If _extensible_ is *false*, return *false*.
             1. Assert: _extensible_ is *true*.
             1. If IsGenericDescriptor(_Desc_) is *true* or IsDataDescriptor(_Desc_) is *true*, then
-              1. If _O_ is not *undefined*, create an own data property named _P_ of object _O_ whose [[Value]], [[Writable]], [[Enumerable]] and [[Configurable]] attribute values are described by _Desc_. If the value of an attribute field of _Desc_ is absent, the attribute of the newly created property is set to its default value.
+              1. If _O_ is not *undefined*, create an own data property named _P_ of object _O_ whose [[Value]], [[Writable]], [[Enumerable]] and [[Configurable]] attribute values are described by _Desc_. If the value of an attribute field of _Desc_ is absent, the attribute of the newly created property is set to its <emu-xref href="#table-default-attribute-values">default value</emu-xref>.
             1. Else,
               1. Assert: ! IsAccessorDescriptor(_Desc_) is *true*.
-              1. If _O_ is not *undefined*, create an own accessor property named _P_ of object _O_ whose [[Get]], [[Set]], [[Enumerable]] and [[Configurable]] attribute values are described by _Desc_. If the value of an attribute field of _Desc_ is absent, the attribute of the newly created property is set to its default value.
+              1. If _O_ is not *undefined*, create an own accessor property named _P_ of object _O_ whose [[Get]], [[Set]], [[Enumerable]] and [[Configurable]] attribute values are described by _Desc_. If the value of an attribute field of _Desc_ is absent, the attribute of the newly created property is set to its <emu-xref href="#table-default-attribute-values">default value</emu-xref>.
             1. Return *true*.
           1. If every field in _Desc_ is absent, return *true*.
           1. If _current_.[[Configurable]] is *false*, then
             1. If _Desc_.[[Configurable]] is present and its value is *true*, return *false*.
-            1. If _Desc_.[[Enumerable]] is present and the [[Enumerable]] fields of _current_ and _Desc_ are the Boolean negation of each other, return *false*.
-          1. If IsGenericDescriptor(_Desc_) is *true*, no further validation is required.
-          1. Else if IsDataDescriptor(_current_) and IsDataDescriptor(_Desc_) have different results, then
+            1. If _Desc_.[[Enumerable]] is present and _Desc_.[[Enumerable]] is different from _current_.[[Enumerable]], return *false*.
+          1. If ! IsGenericDescriptor(_Desc_) is *true*,
+            1. NOTE: No further validation is required.
+          1. Else if ! IsDataDescriptor(_current_) is different from ! IsDataDescriptor(_Desc_), then
```

Comment:
Similarly here, it might be better to do this:
```suggestion
          1. Else if ! SameValue(! IsDataDescriptor(_current_), ! IsDataDescriptor(_Desc_)) is *false*, then
```

Reply:
e19fdba